sessions/sessions.go

```diff
@@ -51,6 +51,8 @@ type SessionProtocol struct {
 	proto    string
 	streams  chan net.Conn
 	sessions sync.Map // types.PublicKey -> *activeSession
+	mutex    sync.Mutex
+	closed   bool
```

Comment:
It might be easier/cleaner to use a `sync.Once` than a `sync.Mutex`+`bool`: https://pkg.go.dev/sync#Once

Reply:
True. My reasoning was that this provides an easy way to check if the session is closed but as that isn't really being used, I've changed it.